src/transforms/onehotencoding.jl

```diff
@@ -0,0 +1,86 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    OneHotEncoding(col)
+    
+Transforms categorical column `col` into one-hot columns of levels
+returned by the `levels` function of CategoricalArrays.jl.
+
+# Examples
+
+```julia
+OneHotEncoding(1)
+OneHotEncoding(:a)
+OneHotEncoding("a")
+```
+"""
+struct OneHotEncoding{S<:ColSelector} <: Stateless
+  col::S
+end
+
+isrevertible(::Type{<:OneHotEncoding}) = true
+
+_colname(col::Integer, names) = names[col]
+_colname(col::AbstractString, names) =
+  _colname(Symbol(col), names)
+
+function _colname(col::Symbol, names)
+  @assert col ∈ names "Invalid column."
+  return col
+end
+
+_name(nm, names) = 
+  nm ∈ names ? _name(Symbol("$(nm)_"), names) : nm
+
+function apply(transform::OneHotEncoding, table)
+  cols = Tables.columns(table)
+  names = collect(Tables.columnnames(cols))
```

Comment:
Do we really need to collect the names? Isn't Tables.columnnames always indexable?

Reply:
Yes, the `splice!` function needs a Vector as the first argument.

Reply:
I would call collect or vec in the splice! function then. Calling it here too early makes the reader of the code wonder why this is necessary.

Reply:
what about splice!(collect(names), ind, newnms)?

Reply:
The `splice!` function returns the removed value, not the transformed vector. So I need a vector in a variable.